internal/auth/ldap/service_authenticate.go

```diff
@@ -0,0 +1,99 @@
+package ldap
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/authtoken"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/iam"
+)
+
+type (
+	// AuthenticatorFactory is used by "service functions" to create a new
+	// ldap.Authenticator (typically an ldap.Repository)
+	AuthenticatorFactory func() (Authenticator, error)
+
+	// LookupUserFactory is used by "service functions" to create a new
+	// LookupUser (typically an iam repo)
+	LookupUserFactory func() (LookupUser, error)
+
+	// AuthTokenCreatorFactory is used by "service functions" to create a new
+	// AuthTokenCreator (typically an auth token repo)
+	AuthTokenCreatorFactory func() (AuthTokenCreator, error)
+)
+
+// Authenticate is an ldap domain service function for handling an LDAP
+// authentication flow. On success, it returns an auth token.
+//
+// The service operation includes:
+//	- Authenticate the user against the auth method's configured ldap server.
+//  - Use iam.(Repository).LookupUserWithLogin(...) look up the iam.User
+//	  matching the Account.
+//  - Use the authtoken.(Repository).CreateAuthToken(...) to create a pending
+//    auth token for the authenticated user.
+
+func Authenticate(
+	ctx context.Context,
+	authenticatorFn AuthenticatorFactory,
+	lookupUser LookupUserFactory,
```

Comment:
nit: should this be lookupUserFn to match the others?

Reply:
ty!